src/wp-includes/kses.php

```diff
@@ -1263,11 +1263,10 @@ function wp_kses_attr_check( &$name, &$value, &$whole, $vless, $element, $allowe
 		 * `data-*` (not to be mixed with the HTML 4.0 `data` attribute, see
 		 * https://www.w3.org/TR/html40/struct/objects.html#adef-data).
 		 *
-		 * Note: the attribute name should only contain `A-Za-z0-9_-` chars,
-		 * double hyphens `--` are not accepted by WordPress.
+		 * @see https://html.spec.whatwg.org/#custom-data-attribute
 		 */
 		if ( str_starts_with( $name_low, 'data-' ) && ! empty( $allowed_attr['data-*'] )
-			&& preg_match( '/^data(?:-[a-z0-9_]+)+$/', $name_low, $match )
+			&& preg_match( '~^data-[^=/> \\t\\f\\r\\n]+$~', $name_low, $match )
```

Comment:
What's the reasoning for matching the inverse of this set of characters specifically? Is this based on the specification for attributes names in HTML?

Reply:
yes exactly, because data attributes themselves are just attributes. there's a large gap of attribute names based on matching ASCII a-z. emoji can form attributes names, for example.

now it might be reasonable to ask, why add this if we don't want to support all those? but the purpose of this detector is to validate data attributes. if we add additional rules above the spec for these, we will miss validating certain data attributes we didn't anticipate.

part of me also didn't want to do this, so maybe what I wrote is wrong, because it ultimately matters most _which of these will be turned into the `dataset` in the browser_ so I'm happy to change this if we think we need to, but it seemed like they _all_ were collected in my testing, and I didn't want to invite escape hatches based on ad-hoc pattern-matching.